gguf-py/tests/test_gguf.py

```diff
@@ -1,7 +1,83 @@
-import gguf  # noqa: F401
+import os
+import struct
+import tempfile
+import unittest
 
-# TODO: add tests
+import numpy as np
+from gguf import GGUFWriter
 
 
-def test_write_gguf() -> None:
-    pass
+class TestGGUFWriter(unittest.TestCase):
+    def setUp(self):
+        self.temp_dir = tempfile.TemporaryDirectory()
+
+    def tearDown(self):
+        self.temp_dir.cleanup()
```

Comment:
Should these functions be `snake_case`?

Reply:
They override unittest metods, and unittest is so old that it predates the standardization on snake_case.